src/shared/Core/Trace2.cs

```diff
@@ -0,0 +1,278 @@
+        }
+
+        var sidSuffix = Guid.NewGuid();
+        var currentSid = $"{parentSid}{sidSuffix}";
+        
+        _context.Environment.SetEnvironmentVariable(CurrentSidVariable, currentSid);
+
+        return currentSid;
+    }
+
+    public void AddListener(CollectorWriter listener)
+    {
+        ThrowIfDisposed();
+        
+        // Try not to add the same listener more than once
+        if (_writer == listener)
+            return;
+
+        _writer = listener;
+        _writer.Start();
+    }
+
+    public new void Dispose()
+    {
+        if (_isDisposed)
+        {
+            // This instance has already been disposed
+            return;
+        }
+        
+        _writer.Dispose();
+        _isDisposed = true;
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed")]
+    public void WriteVersion(
+        string gcmVersion,
+        string eventFormatVersion = "3",
+        [System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
+        [System.Runtime.CompilerServices.CallerLineNumber] int lineNumber = 0)
+    {
+        EnsureArgument.NotNull(gcmVersion, nameof(gcmVersion));
+
+        WriteMessage(new VersionCollectorMessage()
+        {
+            Event = Event.Version,
+            Sid = GetSid(),
+            Time = DateTime.UtcNow.ToString("o"),
+            File = filePath,
+            Line = lineNumber,
+            Evt = eventFormatVersion,
+            Exe = gcmVersion
+        });
+    }
+
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1026:DefaultParametersShouldNotBeUsed")]
+    public void WriteStart(
+        IList<string> argv,
+        double timeInMilliseconds,
+        [System.Runtime.CompilerServices.CallerFilePath] string filePath = "",
+        [System.Runtime.CompilerServices.CallerLineNumber] int lineNumber = 0)
+    {
+        WriteMessage(new StartCollectorMessage()
+            {
+                Event = Event.Start,
+                Sid = GetSid(),
+                Time = DateTime.UtcNow.ToString("o"),
+                File = filePath,
+                Line = lineNumber,
+                Argv = argv,
+                ElapsedTime = timeInMilliseconds
+            });
+    }
+
+    public void WriteExit(
+        int code,
+        double timeInMilliseconds,
+        string filePath = "",
+        int lineNumber = 0)
+    {
+        EnsureArgument.NotNull(code, nameof(code));
+        EnsureArgument.NotNull(timeInMilliseconds, nameof(timeInMilliseconds));
+        
+        WriteMessage(new ExitCollectorMessage()
+        {
+            Event = Event.Exit,
+            Sid = GetSid(),
+            Time = DateTime.UtcNow.ToString("o"),
+            File = filePath,
+            Line = lineNumber,
+            Code = code,
+            // In Git, the "fmt" field is used to write exception information minus any PII.
```

Comment:
I think this comment is left over from a cut/paste.

Reply:
On further examination, this appears to be _both_ an accidental cut 'n paste and an unfinished thought 😆 .